src/util/BlankNodeManager.h

```diff
@@ -87,16 +87,30 @@ class BlankNodeManager {
     // Return true iff the `index` was returned by a previous call to `getId()`.
     bool containsBlankNodeIndex(uint64_t index) const;
 
-   private:
-    // Reserved blocks.
-    std::vector<BlankNodeManager::Block> blocks_;
+    // Merge passed `LocalBlankNodeManager`s to keep alive their reserved
+    // BlankNodeIndex blocks.
+    template <std::ranges::range R>
+    void mergeWith(const R& localBlankNodeManagers) {
+      std::ranges::copy_if(
+          localBlankNodeManagers, std::back_inserter(otherManagers_),
+          [](const std::shared_ptr<const LocalBlankNodeManager>& l) -> bool {
+            return l != nullptr;
+          });
```

Comment:
This implementation has the disadvantage, that cyclicly merging local blank node managers leads to 
endless loops (probably memory leaks + endless loops in the `containsBlankNodeIndex` function.
To guard against this, you would need some other indirect type, that is stores in the vector.
The easiest way would be to make the `Blocks`  a shared_ptr with a custom deleter, and that deleter does the actual deleting of the blocks.
Then you (similar to the local vocab) have a `shared_ptr<Blocks, CustomDeleter>` active blocks, and a `vector<shared_ptr<const Blocks, CustomDeleter>> otherBlocks_`.